packages/priority-queue/src/index.js

```diff
@@ -0,0 +1,50 @@
+const requestIdleCallback = window.requestIdleCallback ? window.requestIdleCallback : window.requestAnimationFrame;
+
+export const createQueue = () => {
+	const waitingList = [];
+	const elementsMap = new WeakMap();
+	let isRunning = false;
+
+	const runWaitingList = ( deadline ) => {
+		do {
+			if ( waitingList.length === 0 ) {
+				isRunning = false;
+				return;
+			}
+
+			const nextElement = waitingList.shift();
+			elementsMap.get( nextElement )();
+			elementsMap.delete( nextElement );
+		} while ( deadline && deadline.timeRemaining && deadline.timeRemaining() > 0 );
+
+		requestIdleCallback( runWaitingList );
+	};
+
+	const add = ( element, item ) => {
+		if ( ! elementsMap.has( element ) ) {
+			waitingList.push( element );
+		}
+		elementsMap.set( element, item );
+		if ( ! isRunning ) {
+			isRunning = true;
+			requestIdleCallback( runWaitingList );
+		}
+	};
+
+	const flush = ( element ) => {
```

Comment:
I don't know why it didn't occur to me previously, but: In this context, wouldn't a `flush` method be expected to run the callback immediately? With this implementation, it acts more like a "delete" than a "flush".

Reply:
I guess the call was being done regardless but on the caller side. Having it here could simplify things.